test/session_test.py

```diff
@@ -257,6 +258,32 @@ def test_raise_error_if_timestamp_is_too_old(self):
             session = shopify.Session("http://localhost.myshopify.com", "unstable")
             session = session.request_token(params)
 
+    def test_access_scopes_are_nil_by_default(self):
+        session = shopify.Session("testshop.myshopify.com", "unstable", "any-token")
+        self.assertIsNone(session.access_scopes)
+
+    def test_access_scopes_when_valid_scopes_passed_in(self):
```

Comment:
Out of curiosity, what will happen if invalid scopes are passed in? Is it worth testing that specifically or can we just trust the ApiAccess tests?

Reply:
There is some test coverage in this file: https://github.com/Shopify/shopify_python_api/blob/455bd5a94d2d29d005798f1b81ba4e3fd816672e/test/api_access_test.py#L13-L17

From the library's perspective, the `ApiAccess` object will raise this error. If a bad access scope is passed in to the permission_url during OAuth, Shopify will elegantly let you know during the redirect dance.

Reply:
I think we can trust the `ApiAccess` tests in here. `ApiAccess` will throw an `ApiAccessError` if that fails. That is already covered in `ApiAccess` tests.